src/Viewer/services/decoder/FILE_FORMATS.js

```diff
@@ -0,0 +1,68 @@
+const FILE_TYPES = Object.freeze({
+    UNKNOWN: "unknown",
+
+    CLP_IR: "clp_ir",
+    GZ: "gz",
+    TAR_GZ: "tar_gz",
+    ZIP: "zip",
+    ZST: "zst",
+});
+
+const FILE_EXTENSION_TO_TYPE = Object.freeze({
+    ".txt": FILE_TYPES.UNKNOWN,
+
+    ".clp.zst": FILE_TYPES.CLP_IR,
+    ".gz": FILE_TYPES.GZ,
+    ".gzip": FILE_TYPES.GZ,
+    ".tar.gz": FILE_TYPES.TAR_GZ,
+    ".zip": FILE_TYPES.ZIP,
+    ".zst": FILE_TYPES.ZST,
+});
+
+const FILE_TYPE_FULL_NAMES = Object.freeze({
+    [FILE_TYPES.UNKNOWN]: "Plain text",
+
+    [FILE_TYPES.CLP_IR]: "CLP IR stream",
+    [FILE_TYPES.GZ]: "Gzip",
+    [FILE_TYPES.TAR_GZ]: "Tarball Gzip",
+    [FILE_TYPES.ZIP]: "ZIP",
+    [FILE_TYPES.ZST]: "Zstandard",
+});
+
+/* eslint-disable no-magic-numbers, @stylistic/js/array-element-newline */
+const FILE_TYPE_MAGIC_NUMBERS = Object.freeze({
```

Comment:
Can we alphabetize the keys here as well? If you're concerned about the comment for tar.gz being above the comment from gz, we could flip the comments as well so that gz has the complete description and tar.gz has the incremental description.